### PR TITLE
Apply static troposphere model when weather model file is not provided

### PR DESCRIPTION
**Pull request by seongsujeong**
This PR is to add a feature that applies static troposphere when weather model was not provided in runconfig. In case the weather model file is provided, the tropospheric correction is based on the weather model.
The static troposphere LUT can be calculated from incidence angle and surface height, which are calculated in `lut.py`.

### CODE REVIEW

**Review by LiangJYu — APPROVED** [review]
LGTM

File: src/compass/utils/h5_helpers.py [hunk 1]
```diff
@@ -617,7 +617,13 @@ def corrections_to_h5group(parent_group, burst, cfg, rg_lut, az_lut,
                  f'Ionospheric delay (range) {desc}',
                  {'units': 'meters'}),
         ]
-        if weather_model_path is not None:
+
+        if weather_model_path is None:
+            correction_items.append(Meta('static_los_troposphere_delay',
+                                             ds.GetRasterBand(5).ReadAsArray(),
+                                             f'Static LOS troposphere delay {desc}',
```
Comment (vbrancat) [inline src/compass/utils/h5_helpers.py:624]: I would not include the static troposphere as a correction layer inside the product. This correction can be easily calculated using the static layers product. So, there should be no need to have it there too

File: src/compass/utils/lut.py [hunk 2]
```diff
@@ -440,3 +449,27 @@ def resample_set(geo_tide, pts_src, pts_dest):
...
     return rdr_tide
+
+
+def compute_static_troposphere_delay(incidence_angle_arr, hgt_arr):
+    '''
+    Compute troposphere delay using static model
+
+    Parameters:
+    -----------
+    inc_path: str
```
Comment (vbrancat) [inline src/compass/utils/lut.py:460]: let's specify that the incidence angle is expected in degrees

**Review by vbrancat — COMMENTED** [review]
Thanks you @seongsujeong. I am not sure if we should include the static troposphere inside the product. This correction can be easily computed using the static layers product. Let's discuss in more details

File: tests/test_s1_geocode_slc.py [hunk 3]
```diff
@@ -117,4 +117,4 @@ def test_geocode_slc_validate(geocode_slc_params):
 
     # check for bright spots in sliced array
     corner_reflector_threshold = 3e3
-    assert np.any(arr > corner_reflector_threshold)
+    assert np.any(np.abs(arr) > corner_reflector_threshold)
```
Comment (seongsujeong) [inline tests/test_s1_geocode_slc.py:120]: Here is the suggested change in the test code. 
It looks like `complex64` is being compared to `float`. In this case, it seems numpy only takes the real part of the complex number and to the comparison. I think it makes more sense to calculate the magnitude of the complex image. @LiangJYu What do you think?


> np.array([3001 + 9000 * 1.0j]) > 3e3
array([ True])
> np.array([2001 + 9000 * 1.0j]) > 3e3
array([False])
> np.array([2001 + 1000 * 1.0j]) > 3e3
array([False])
> np.array([3001 + 1000 * 1.0j]) > 3e3
array([ True])

File: src/compass/utils/h5_helpers.py (hunk 1 above)
Comment (seongsujeong) [reply src/compass/utils/h5_helpers.py]: Got it. I've changed the static troposphere LUT not to be written into the product.

File: src/compass/utils/h5_helpers.py [hunk 4]
```diff
@@ -614,16 +614,19 @@ def corrections_to_h5group(parent_group, burst, cfg, rg_lut, az_lut,
         Meta('los_solid_earth_tides', ds.GetRasterBand(4).ReadAsArray(),
              f'Solid Earth tides (range) {desc}',
              {'units': 'meters'}),
+        Meta('los_ionospheric_delay', ds.GetRasterBand(5).ReadAsArray(),
```
Comment (seongsujeong) [inline src/compass/utils/h5_helpers.py:617]: NOTE: The most recent branch as of now does not seem to have ionosphere LUT written into the product, which seem to happen on #112. Here I add the ionosphere back. Also note that the band ID of the LUTs are adjusted accordingly.

File: tests/test_s1_geocode_slc.py (hunk 3 above)
Comment (LiangJYu) [reply tests/test_s1_geocode_slc.py:120]: Good find! Please implement your proposed fix.

**Review by vbrancat — APPROVED** [review]
Thank you for having addressed my comments. LGTM

**Pull request merged**